doc/api/async_hooks.md

```diff
@@ -460,6 +460,57 @@ init for PROMISE with id 6, trigger id: 5  # the Promise returned by then()
   after 6
 ```
 
+#### async_hooks.executionAsyncResource()
+
+<!-- YAML
+added: REPLACEME
+-->
+
+* Returns: {Object} The resource that triggered the current
+  execution context.
+  Useful to store data within the resource.
```

Comment:
It should be documented here that most often, this object is an internal object that has an undocumented internal API that may change at any time and that attempting to use any functions on the object may actually cause the node.js process to crash if called inappropriately.

Reply:
Added a doc note about that. :+1: